### PR TITLE
Use WP Autoloader class

### PR DESCRIPTION
**Pull request by obenland**
This is one of those changes I'm on the fence about as it doesn't strictly improve things, it just makes them "different".
The Autoloader class proposed here is based on [Dion's work for the WP.org Plugin Directory](https://meta.trac.wordpress.org/browser/sites/trunk/wordpress.org/public_html/wp-content/plugins/plugin-directory/class-autoloader.php?rev=3083) and used in its proposed form in the `/lib/atomic-v2` library on Dotcom.

I was positively surprised that it seems to just be a drop-in replacement for the current Autoloader, which I didn't quite expect :)
It currently lacks the debug log for when a class is not readable or couldn't be loaded. Was that triggered at any point before?

## Proposed changes:

* Replaces inline Autoloader with a separate Autoloader class that can be re-used.
* Moves the integrations loader to be included with the other includes.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?

## Testing instructions:

* Make sure everything still works?

### CODE REVIEW

File: activitypub.php [hunk 1]
```diff
@@ -161,8 +126,6 @@ function plugin_settings_link( $actions ) {
 	)
 );
 
-// Load integrations.
-require_once __DIR__ . '/integration/load.php';
```
Comment (pfefferle) [inline activitypub.php:148]: Can we delete this? It is not the included folder!?!

Comment (obenland) [reply activitypub.php:148]: It's moved to [line 34](https://github.com/Automattic/wordpress-activitypub/pull/1051/files#diff-d33bfd446d4ad05f88e9c8e907f31b2e4c06eca40c7f28cee7e58f3a2302958bR34), to group with the other includes.

Comment (pfefferle) [reply activitypub.php:148]: Ah, haven't seen that! Thanks!

**Review by pfefferle — DISMISSED** [review]
I like it and it looks way cleaner

File: includes/class-autoloader.php [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+	/**
+	 * Registers Autoloader's autoload function.
+	 *
+	 * @throws \Exception When autoload_function cannot be registered.
+	 *
+	 * @param string $prefix Namespace prefix all classes have in common.
+	 * @param string $path   Path to the files to be loaded.
+	 */
+	public static function register_path( $prefix, $path ) {
+		$loader = new Autoloader( $prefix, $path );
```
Comment (pfefferle) [inline includes/class-autoloader.php]: Maybe use `self` if the file gets renamed sometime?

```suggestion
		$loader = new self( $prefix, $path );
```

**Comment by pfefferle** [comment]
Maybe escaping the global functions!?!

File: activitypub.php [hunk 3]
```diff
@@ -81,6 +85,22 @@ function plugin_init() {
 }
 \add_action( 'plugins_loaded', __NAMESPACE__ . '\plugin_init' );
 
+/**
+ * Add plugin settings link.
+ *
+ * @param array $actions The current actions.
+ */
+function plugin_settings_link( $actions ) {
```
Comment (pfefferle) [inline activitypub.php]: haven't we moved that to the `class-admin.php`?

**Comment by obenland** [comment]
> Maybe escaping the global functions!?!

What do you mean with that?

**Comment by pfefferle** [comment]
Why have I used "escape" and I have no idea how it is called 🫣... I mean `\spl_autoload_register` instead of `spl_autoload_register`.

**Review by pfefferle — APPROVED** [review]

**Pull request merged**